x/subscription/types/message_add_project.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const TypeMsgAddProject = "add_project"
+
+var _ sdk.Msg = &MsgAddProject{}
+
+func NewMsgAddProject(creator string, projectName string, enable bool, consumer string, vrfpk string) *MsgAddProject {
+	return &MsgAddProject{
+		Creator:     creator,
+		Consumer:    consumer,
+		ProjectName: projectName,
+		Enable:      enable,
+		Vrfpk:       vrfpk,
+	}
+}
+
+func (msg *MsgAddProject) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgAddProject) Type() string {
+	return TypeMsgAddProject
+}
+
+func (msg *MsgAddProject) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgAddProject) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgAddProject) ValidateBasic() error {
```

Comment:
validate: 
project name (non empty and max length?)
creator + consumer are account addreses (format)?